src/nvim/cursor_shape.c

```diff
@@ -77,7 +81,8 @@ Array mode_style_array(void)
   return all;
 }
 
-/// Parse the 'guicursor' option
+/// Parse the 'guicursor' option, or clears `shape_table` if 'guicursor'
```

Comment:
I think doxygen allows to reference symbols via `\ref` so you could write `\ref  shape_table`.